Tasks/Ignatchik/Additional_task/Task_1/search_and_close_tag.py

```diff
@@ -0,0 +1,156 @@
+import re
+
+with open("input.html", 'r', encoding='utf-8') as file:
+    list_html = file.readlines()
+
+flag = True
+html_line = ""
+list_of_all_tags = []
+pattern = '<(\S*?)[^>]*>.*?|<.*?/>'
+
+with open("input.html", "r") as input_file:
+    for line in input_file:
+        html_line += line
+    
+all_tags = re.finditer(pattern, html_line)
+
+try:
+    while flag:
+        item = next(all_tags)
+        list_of_all_tags.append(item.group())
+except StopIteration:
+    flag = False
+
+single_HTML_tags = ['!DOCTYPE', 'area', 'base', 'br', 'col', 'embed', 'hr', 'img', 'input', 'keygen', 'link', 'meta', 'param', 'source', 'track', 'wbr']
+
+def remove_text(l):
+    c_tags = []
+    for tag in l:
+        new_tag = tag.split(" ")
+        if len(new_tag) > 1:
+            c_tags.append(f"{new_tag[0]}>")
+        else:
+            c_tags.append(new_tag[0])
+    return c_tags
+
+def remove_single_tags(l1, l2):
+    res = []
+    for i in l1:
+        count = 0
+        for j in l2:
+            if j not in i:
+                count += 1
+        if count == 16:
```

Comment:
why 16?

Reply:
Because I know 16 single tags. It would have been better to write len(single_HTML_tags)?